src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -76,12 +87,25 @@ public void visitToken(DetailAST ast) {
             final DetailAST switchAST = defaultGroupAST.getParent();
             final DetailAST lastGroupAST =
                 switchAST.getLastChild().getPreviousSibling();
-
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
-                || defaultGroupAST.getColumnNo()
-                    != lastGroupAST.getColumnNo()) {
+            if (!skipIfLastAndSharedWithCaseInSwitch(ast)
+                    && (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
```

Comment:
@rnveach ,

due to blame history it was in original code (13 year ago) - https://github.com/checkstyle/checkstyle/blame/836bb79348590f1a473f66553ea2ce3c9dec2786/src/checkstyle/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

So please move that to separate issue to figure-out , it might be just very old code.